src/parse.yy

```diff
@@ -1667,7 +1681,26 @@ initializer_list
       }
     ;
 
+attribute_list
```

Comment:
Should we chain pragmas? We accept only one pragma and have not check for pragma compatibility.
I don't care about pragma compatibility much, if it doesn't crash the program. But the following example does:
```c
void foo();
void bar() {
#pragma nounroll
#pragma unroll (8)
  for(uniform int i = 0; i<100; i++) {
    foo();
  }
}
```

Reply:
There is a mistake there. But I think the right way to chain would be in grammar rule for statement. Sfould be:

statement 
       : attribute_list statement
         { ....}
        | statement_base
       ;
This will allow us to handle new pragmas if added here and chain them. In this particular case, throw error since we are trying to set opp unroll commands for same loop.

Reply:
Agree. Also probably better names here are:
statement_base -> statement
statement -> attributed_statement

Reply:
Done + change made to identify conflicting chaining of loop pragmas. The conflict will be identified in setloopattribute()